start-utils

```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+function join_by { local d=$1; shift; echo -n "$1"; shift; printf "%s" "${@/#/$d}"; }
```

Comment:
Super cool!

Reply:
```
 thing=(abc def 123)
$ printf "%s|" ${thing[@]} | sed 's/.$//'
abc|def|123
```
You've basically cramed 5 lines into 1. 

If you're case you could do
``` 
join(){ 
   local d=${1:?You need a delimiter} 
   printf "%s$d" "${@:1}" | sed 's/.$//'; 
}
```
the `?` in the variable assignment is an assertion - probably a good idea